lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -53,6 +53,16 @@ module AutoEncrypter
       def setup_encrypter(options = {})
         opts = set_default_options(options.dup)
 
+        unless opts[:key_vault_client]
+          # If no key vault client is passed in, create one by copying the
+          # Mongo::Client used for encryption. Update options so that key vault
+          # client does not perform auto-encryption/decryption, and keep a reference
+          # to it so it is destroyed later.
+          @key_vault_client = self.with({ auto_encryption_options: nil })
```

Comment:
Could you also add a test to client_construction_spec.rb which illustrates whether changing AEO creates a new cluster.

Reply:
And whether it is possible to use the same client instance to access key vault, since auto encryption should be completely governed by the schema and the key vault collection is separate from user collections (?) such that the schema should not specify encryption for the key vault collection.